packages/scene-composer/src/components/panels/ComponentEditMenu.tsx

```diff
@@ -0,0 +1,168 @@
+});
+
+const textStrings = defineMessages({
+  [ObjectTypes.AddDataBinding]: { defaultMessage: 'Add data binding', description: 'Menu Item' },
+  [ObjectTypes.RemoveAllDataBinding]: { defaultMessage: 'Remove all data binding', description: 'Menu Item' },
+  [ObjectTypes.RemoveOverlay]: { defaultMessage: 'Remove overlay', description: 'Menu Item' },
+});
+
+export const ComponentEditMenu: React.FC<ComponentEditMenuProps> = ({ nodeRef, currentComponent }) => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const updateComponentInternal = useStore(sceneComposerId)((state) => state.updateComponentInternal);
+  const removeComponent = useStore(sceneComposerId)((state) => state.removeComponent);
+  const { formatMessage } = useIntl();
+
+  const mapToMenuItem = useCallback(
+    (item: ToolbarItemOptionRaw): ComponentEditMenuItem => {
+      const typeId: ObjectTypes = item.uuid as ObjectTypes;
+      return {
+        ...item,
+        uuid: typeId,
+        subItems: item.subItems?.map(mapToMenuItem),
+        label: formatMessage(labelStrings[typeId]),
+        text: textStrings[typeId] ? formatMessage(textStrings[typeId]) : undefined, // if there's a text string, show text
+      };
+    },
+    [formatMessage],
+  );
+
+  const componentItems = useMemo(() => {
+    switch (currentComponent.type) {
+      case KnownComponentType.DataBinding:
+        return [
+          {
+            uuid: ObjectTypes.AddDataBinding,
+          },
+          {
+            uuid: ObjectTypes.RemoveAllDataBinding,
+          },
+        ];
+
+      case KnownComponentType.DataOverlay: {
+        const items = [
+          {
+            uuid: ObjectTypes.AddDataBinding,
+          },
+        ];
+        if ((currentComponent as IDataOverlayComponentInternal).subType === Component.DataOverlaySubType.OverlayPanel) {
+          items.push({
+            uuid: ObjectTypes.RemoveOverlay,
+          });
+        }
+        return items;
+      }
+      default:
+        return [];
+    }
+  }, [currentComponent]);
+
+  const menuItems = useMemo(
+    () =>
+      [
+        {
+          icon: { name: 'ellipsis' as IconProps.Name },
+          uuid: ObjectTypes.EditComponent,
+        },
+        ...componentItems,
+      ].map(mapToMenuItem),
+    [currentComponent],
+  );
+
+  const handleAddDataBinding = useCallback(() => {
+    switch (currentComponent.type) {
+      case KnownComponentType.DataBinding: {
+        const newComponentPartial = {
+          ...currentComponent,
+          valueDataBindings: [...((currentComponent as IDataBindingComponentInternal).valueDataBindings || []), {}],
+        };
+        updateComponentInternal(nodeRef, newComponentPartial);
+        return;
+      }
+      case KnownComponentType.DataOverlay: {
+        const newComponentPartial = {
+          ...currentComponent,
+          valueDataBindings: [
+            ...((currentComponent as IDataOverlayComponentInternal).valueDataBindings || []),
```

Comment:
Nit: Consider how to avoid type assertion.

Reply:
as I've mentioned previously, I don't know if there is a way to avoid this kind of type assertion given our current setup

Reply:
this is a new menu icon added to inspector panel. we have 4 kinds of valueDataBindings right now. don't expect to seeing additional ones in the future and I may revisit that soon, but at least we will still need to check the component type to see which kind of valueDataBindings the current component is using